File: 01/solution.py
Data = [int(x) for x in open("./data.txt", "r").read().split("\n")]

# first part
Frequency = sum(Data)
print(Frequency)

#second part
Frequency = 0
Duplicates = []

i = 0

while True:
     Frequency += Data[i]
     if Frequency in Duplicates:
         break
     Duplicates.append(Frequency)
     i = (i + 1) % len(Data)

print(Frequency)

File: 01/solution.rb
data = File.open("data.txt")
                .read()
                .split("\n")
                .map(&:to_i)

# first part
puts "First part: " + data.inject(0, &:+).to_s

# second part
duplicates = Hash.new
$freqency = 0

data.cycle do |x|
    $freqency += x
    break if duplicates.has_key?($freqency)
    duplicates.store($freqency, TRUE)
end

puts "Second part: " + $freqency.to_s